sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionTests.java

```diff
@@ -0,0 +1,616 @@
+                 array_repeat
+                --------------
+                 {}
+                (1 row)
+                
+                SELECT array_repeat(null, -1);
+                 array_repeat
+                --------------
+                 {}
+                (1 row)
+                
+                SELECT array_repeat(null, 3);
+                 array_repeat
+                --------------
+                 {null,null,null}
+                (1 row)
+                """
+        );
+    }
+
+    @Test @Ignore("ICE: cannot add expression of different type to set: INTEGER NOT NULL ARRAY NOT NULL -> INTEGER NOT NULL ARRAY")
+    public void testArrayRepeat2() {
+        this.qs("""
+                SELECT array_repeat(123, null);
+                 array_repeat
+                --------------
+                 NULL
+                (1 row)
+                """
+        );
+    }
+
+    @Test @Ignore("ICE: compilation failure while optimizing")
+    public void testArrayRepeat3() {
+        this.qs("""
+                SELECT array_repeat('a', 6);
+                 array_repeat
+                --------------
+                 { a, a, a, a, a, a}
+                (1 row)
+                """
+        );
+    }
+
+    @Test
+    public void testArrayRemove() {
+        this.qs("""
+                SELECT array_remove(ARRAY [2, 2, 6, 6, 8, 2], 2);
+                 array_remove
+                --------------
+                 {6, 6, 8}
+                (1 row)
+                
+                SELECT array_remove(ARRAY [1, 2, 3], null);
+                 array_remove
+                --------------
+                 NULL
+                (1 row)
+                
+                SELECT array_remove(ARRAY [null, 2, 2, 6, 6, 8, 2], 2);
+                 array_remove
+                --------------
+                 {null, 6, 6, 8}
+                (1 row)
+                
+                SELECT array_remove(ARRAY [null, 2, 2, 6, 6, 8, 2], null);
+                 array_remove
+                --------------
+                 NULL
+                (1 row)
+                
+                SELECT array_remove(ARRAY [2, 2, 6, 6, 8, 2], elem) FROM (SELECT elem FROM UNNEST(ARRAY [2, 6, 8]) as elem);
+                 array_remove
+                --------------
+                 {6, 6, 8}
+                 {2, 2, 8, 2}
+                 {2, 2, 6, 6, 2}
+                (3 rows)
+                
+                SELECT array_remove(ARRAY [2, 2, 6, 6, 8, 2], elem) FROM (SELECT elem FROM UNNEST(ARRAY [2, 6, 8, null]) as elem);
+                 array_remove
+                --------------
+                 {6, 6, 8}
+                 {2, 2, 8, 2}
+                 {2, 2, 6, 6, 2}
+                 NULL
+                (4 rows)
+                
+                """
+        );
+    }
+
+    @Test @Ignore("TBD: Should we support implicitly casting here? Snowflake thinks its fine as long as types are comparable")
```

Comment:
Yes, similar to other functions on arrays, the element here should be considered cast to the type of the array element